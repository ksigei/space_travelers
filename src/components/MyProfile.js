import React from 'react';
import { useSelector } from 'react-redux';

function MyProfile() {
  /* eslint-disable */ 
  const reservedRockets = useSelector((state) => state.rocketReducers.filter((rocket) => rocket.rocketBooked));
  /* eslint-enable */
  return (
    <div>
      {reservedRockets.map((rocket) => (
        <div key={rocket.id}>
          <h3>{rocket.rocketName}</h3>
        </div>
      ))}
    </div>
  );
}

export default MyProfile;
