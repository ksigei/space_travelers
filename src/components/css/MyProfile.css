.reserved-box {
  width: 100%;
  display: flex;
}

.reserved-box .missions,
.reserved-box .rockets {
  flex: 1;
  margin: 0 20px;
}

.mission,
.rocket {
  border: 1px solid #e4e4e4;
  padding: 7px;
  font-weight: 600;
}

h3 {
  margin-bottom: 7px;
}
