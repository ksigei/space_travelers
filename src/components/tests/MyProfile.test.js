import React from 'react';
import renderer from 'react-test-renderer';
import '../MyProfile';

test('check if we have h3 tag', () => {
  const tree = renderer.create(
    <h3>My Missions</h3>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
