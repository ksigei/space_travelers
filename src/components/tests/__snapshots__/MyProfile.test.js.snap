// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<h3>
  My Missions
</h3>
`;

exports[`check if we have h3 tag 1`] = `
<h3>
  My Missions
</h3>
`;

exports[`matches snapshot 1`] = `
<tr>
  <th>
    Mission
  </th>
  <th>
    Description
  </th>
  <th>
    Status
  </th>
</tr>
`;
